src/foam/nanos/approval/ApprovalTester.java

```diff
@@ -59,8 +59,7 @@
    * @return approval status of the test
    */
   public ApprovalStatus test(ApprovalRequest request) {
-    if ( request.getStatus() == ApprovalStatus.REQUESTED
-      || incrRejected(request) >= request.getRequiredRejectedPoints()
+    if ( incrRejected(request) >= request.getRequiredRejectedPoints()
```

Comment:
would check `request.getStatus() == ApprovalStatus.REQUESTED` after the rejection check instead of removing
```suggestion
    if ( incrRejected(request) >= request.getRequiredRejectedPoints()
      || request.getStatus() == ApprovalStatus.REQUESTED
```

Reply:
would this lead to the same result though? if the first approvalrequest is status requested, then `request.getStatus()` will still be returned

Reply:
Well yeah, it's an OR operation the order doesn't matter. But this `test` method is to early return when it see a REQUESTED or adequate rejection point collected.

Since the ApprovalDAO would cleanup pending approval requests once rejected/approved, the ApprovalTester class doesn't expect the sequence [REQUESTED, ..., REJECTED] where you have a rejection and there remain pending approval requests before it.
https://github.com/foam-framework/foam2/blob/c09f7b86068250a7cad45bd27ee0a234709870e6/src/foam/nanos/approval/ApprovalDAO.js#L61

Reply:
i see now.. actually the problem here is that the approvalrequest are different classifications so it is not getting cleaned up. But the dao i passed into getstate does not filter the classification. 
I will close this pr and re-fix, thanks!